include/triton/core/tritonserver.h

```diff
@@ -894,6 +894,25 @@ TRITONSERVER_InferenceTraceSpawnChildTrace(
     struct TRITONSERVER_InferenceTrace* trace,
     struct TRITONSERVER_InferenceTrace** child_trace);
 
+/// Set TRITONSERVER_InferenceTrace context.
+///
+/// \param trace The trace.
+/// \param trace_context A new trace context to associate with the trace.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_DECLSPEC struct TRITONSERVER_Error*
+TRITONSERVER_InferenceTraceSetContext(
+    struct TRITONSERVER_InferenceTrace* trace, const char* trace_context);
+
+
+/// Get TRITONSERVER_InferenceTrace context.
+///
+/// \param trace The trace.
+/// \param trace_context Returns the context associated with the trace.
+/// \return a TRITONSERVER_Error indicating success or failure.
+TRITONSERVER_DECLSPEC struct TRITONSERVER_Error*
+TRITONSERVER_InferenceTraceContext(
```

Comment:
Just for my own understanding, if this function is called exclusively by the Python backend (at least on this change), should it be placed at the `tritonbackend.h`?

Reply:
This can be also potentially be called from the front-end if there is a need to update context, i.e.
- context = TRITONSERVER_InferenceTraceContext
- Add value to existing (not `""`) context -> updated_context = context.addEntry(...)
- TRITONSERVER_InferenceTraceSetContext(triton_trace, updated_context)

Reply:
However, I do see your point and it might be reasonable to move `TRITONSERVER_InferenceTraceContext` to `tritonbackend.h` 

I don't think that a hypothetical scenario described above is valuable.

Reply:
Thanks, that makes sense. It looks like we are keeping all the Trace and Metrics APIs in the `tritonserver.h` anyway.